File: examples/test-projects/small-projects/test-project-php/FileStorage.php
<?php
/** Файловое хранилище — аналог FileStorage из других малых проектов. */

class FileStorage {

    public function saveCounter(int $value): void {
        $f = fopen('counter.dat', 'w');
        fwrite($f, (string)$value);
        fclose($f);
    }

    public function loadCounter(): int {
        $f = fopen('counter.dat', 'r');
        $value = (int)fread($f, 64);
        fclose($f);
        return $value;
    }

    public function appendLog(string $message): void {
        $f = fopen('app.log', 'a');
        fwrite($f, $message . "\n");
        fclose($f);
    }

    public function readLog(): string {
        $f = fopen('app.log', 'r');
        $line = fgets($f);
        fclose($f);
        return $line !== false ? $line : '';
    }
}

File: examples/test-projects/small-projects/test-project-cpp/file_storage.cpp
#include "file_storage.h"

#include <fstream>
#include <sstream>
#include <string>

// Запись значения в файл "counter.dat" через выходной файловый поток.
void saveCounter(int value) {
    std::ofstream ofs("counter.dat");
    ofs << value << std::endl;
    ofs.close();
}

// Чтение значения из файла "counter.dat" через входной файловый поток.
int loadCounter() {
    std::ifstream ifs("counter.dat");
    int value = 0;
    ifs >> value;
    ifs.close();
    return value;
}

// Дозапись сообщения в журнал "app.log".
void appendLog(const std::string& message) {
    std::ofstream log("app.log", std::ios::app);
    log << message << std::endl;
    log.close();
}

// Чтение всего журнала "app.log".
std::string readLog() {
    std::ifstream log("app.log");
    std::stringstream buffer;
    buffer << log.rdbuf();
    log.close();
    return buffer.str();
}
